exercise/submission_models.py

```diff
@@ -39,7 +51,7 @@ def create_from_post(self, exercise, submitters, request):
             logger.exception("Failed to save submitted files: %s %s",
                 request.user.username, exercise);
             new_submission.delete()
-            return None
+            raise DatabaseError
```

Comment:
I have a feeling that this instantiates a new DatabaseError instead or re-raising the caught DatabaseError. It's better to re-use the existing error. It contains all the details added by Django when it raised the error.

Reply:
Ok, should the details be used somewhere or are the current messages fine?

Reply:
I think that `raise DatabaseError from error` also instantiates a new DatabaseError object. When you have the error variable referring the original exception, you can `raise error` and then it certainly does not create a new exception object.

I don't know if these errors have such messages that we could show them directly to the users. If they are descriptive, they could be shown to the user. Just saying "some error" is not helpful even though the old message is like that. It is also important to log errors so that sysadmins can inspect them afterwards. If the `Submission.objects.create` method fails, the current code does not seem to even log the error, so that could be added.

Reply:
just `raise` or `raise NewException(message) from error` would be good. The Latter if the type or the message should be changed.